modules/indicia_setup/db/version_6_1_0/202107151745_list_text_value_of_float_attribute.sql

```diff
@@ -0,0 +1,590 @@
+-- For each view, `list_*_attribute values`, alter the `value` field
+-- for float data types to return the `text_value`, if there is one.
+-- The text value is the entered value, including trailing zeroes,
+-- which we need when a decimal validation rule is in place checking
+-- the number of digits after the decimal place.
+-- Decimal validation and ranges do not play nicely together.
+
+-- I.e.
+-- 
+-- WHEN 'F'::bpchar THEN av.float_value::text ||
+-- CASE
+--     WHEN (a.data_type = ANY (ARRAY['I'::bpchar, 'F'::bpchar])) AND a.allow_ranges = true THEN COALESCE(' - '::text || av.upper_value::text, ''::text)
+--     ELSE ''::text
+-- END
+--
+-- becomes
+--
+-- WHEN 'F'::bpchar THEN
+-- CASE
+--     WHEN av.text_value IS NOT NULL THEN av.text_value
+--     ELSE av.float_value::text ||
+--     CASE
+--         WHEN (a.data_type = ANY (ARRAY['I'::bpchar, 'F'::bpchar])) AND a.allow_ranges = true THEN COALESCE(' - '::text || av.upper_value::text, ''::text)
+--         ELSE ''::text
+--     END
```

Comment:
The entered float or float range is always copied in to the text value so this can be simplified.

Reply:
Wouldn't that mean we'd need to run a script to update existing data?

Reply:
Ah, yes, I remember now, that's why I left that in. Thanks.
What is your preference in general: to have more complex code to support legacy data or to update existing data?

Reply:
In general, if the data can be updated relatively easily, then I'd prefer to see the code base kept as simple as possible.

Reply:
Query to update data and simplified views to list attributes are in commit https://github.com/Indicia-Team/warehouse/commit/b138dbcafdad15863cd4bf71664df6716234faff